fetch_images.py

```diff
@@ -34,12 +34,77 @@
 COPERNICUS_PASS = os.environ["COPERNICUS_PASS"]
 
 
+class linear_function:
+    m = 0
+    b = 0
+
+    def __init__(self, lon1, lat1, lon2, lat2):
+        self.lon1 = lon1
+        self.lat1 = lat1
+        self.lon2 = lon2
+        self.lat2 = lat2
+        self.m = (lat2-lat1) / (lon2-lon1)
+        self.b = lat2 - m * lon2
+
+    def intersect(self, line):
+        x = (self.b - line.b) / (self.m - line.m)
+        y = self.m * x + self.b        
+        if((x>=min(lon1,lon2)&&x<=max(lon1,lon2)) && (y>=min(lat1,lat2)&&y<=max(lat1,lat2))):
+            return (x, y)
+        
+            
+       
+
 def bounds_to_points(bound_box):
     lon1, lat1, lon2, lat2 = bound_box
     return [
         (lon1, lat1), (lon2, lat1), (lon2, lat2), (lon1, lat2), (lon1, lat1)
     ]
 
+def corine_class_in_area (bound_box_points, corine_data):
```

Comment:
Have you tried to do intersections with http://toblerity.org/rtree/ ?
I used it initially for intersecting footprint polygons.

I think it would make the code much more simple, if you created a single polygon for a tile and intersected it against an rtree containing all corine patches.